src/Data/YAML/Dumper.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE Safe              #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module Data.YAML.Dumper
+    ( dumpYAML
+    , dumpYAML'
+    , dumpEvents
+    ) where
+
+import           Data.YAML.Event.Internal as YE
+import           Data.YAML.Internal       as YI
+import           Data.YAML.Schema         as YS
+import           Data.YAML.Event.Writer   (writeEvents)
+
+import qualified Data.ByteString.Lazy     as BS.L
+import           Data.Map                 as Map
+import qualified Data.Text                as T
+
+
+type EvList = [Either String Event]
+type Node2EvList = [Node ()] -> EvList
+
+-- | Dump YAML Nodes using specified UTF encoding to a lazy 'BS.L.ByteString'
+--
+-- @since 0.2.0
+dumpYAML ::  Encoding -> [Node ()] -> BS.L.ByteString
```

Comment:
so this is the dual to `decodeNode`; I'd expect this rather to have a type-sig like

```hs
encodeNode ::  [Doc (Node ())] -> BS.L.ByteString
```

I'd omit the Encoding and default to UTF8, as this is basically supposed to be a convenience function which encodes using the "YAML 1.2 Core schema" and with UTF-8 (which is likely the most common encoding used for YAML files)